packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/LegacyManifest.kt

```diff
@@ -106,11 +106,11 @@ class LegacyManifest private constructor(
           Date()
         }
       }
-      var runtimeVersion = rawManifest.getSDKVersion()
-      // Use the manifest's runtimeVersion if it is not null AND the config's runtimeVersion is not null
-      val rawRuntimeVersion = rawManifest.getRuntimeVersion()
-      if (rawRuntimeVersion != null && configuration.runtimeVersion != null) {
-          runtimeVersion = rawRuntimeVersion
+
+      val runtimeVersion = if (rawManifest.getRuntimeVersion() != null) {
```

Comment:
What if the config's runtimeVersion is null? Do we want to still use the manifest's runtime version or preserve the logic prior to this commit?

Reply:
Now that we are serving two different updates this should never happen. The config.runtimeVersion check was only necessary when the same sdk+rtv manifest was served to both sdk and rtv builds.

 * If the config has no runtimeVersion, the manifest request will have only an expo-sdk header and the response will have a manifest that only contains an sdk version.
(Conversely, if the config has a runtime, we return a manifests with both an sdk and rtv field and use the rtv field exclusively as the correct runtime).

A client without a config.runtimeVersion getting a sdk+rtv manifest should never happen. 

Q: OK, so we don't *need* the config.runtimeVersion check anymore, but isn't it safer in this rare failure case? 
 * A: Falling back to the sdkVersion, *compounds* the bad "this should not be happening" situation by introducing ambiguity: _the row we save in the client database is no longer uniquely defined by the received manifest_ i.e. we can have rows with the same primary key, but different values for their `version` column. Thus we are in back in the situation where the "primary key is sufficient for equality and uniqueness of updates" assumption is violated and we get buggy situations like the motivating "updates saved with sdk version blocking a client from refreshing after an upgrade to a rtv build" edge case.

Reply:
I agree with @jkhales here -- we should rely on the server to respect the headers we're sending and respond with the correct update (including `runtimeVersion` if and only if the config has a runtime version <=> expo-runtime-version header is defined in the request).

If the server responds with an update that doesn't match our config (e.g. an update with `runtimeVersion` defined even if our config doesn't have a runtime version), we should treat it as incompatible rather than reading the sdk version -- because if we read the sdk version we're again breaking the assumption that an update only has one compatible version, and we can run into the same edge case as before.

Reply:
I think that makes sense. I'm thinking about the scenario with self-hosted updates and a static web server.

However, it sounds like in that case we never put an app into a bad state. If a manifest contains both the RTV and SDK version and an `expo build` app (with only SDK version, no custom RTV) downloads the manifest, this code will look at only the RTV in the manifest and then decide not to launch the update because the custom RTV doesn't match the app's SDK version. So, the app won't run the new update and will just launch the new one.

That sounds like a good failure mode because for all we know, the update with the custom RTV maybe truly needed custom native code and would be unsafe to run in an app with only a preset SDK.